compaction/compaction_manager.cc

```diff
@@ -277,16 +279,17 @@ compaction_manager::task::task(compaction_manager& mgr, compaction::table_state*
     , _description(std::move(desc))
 {}
 
-future<> compaction_manager::perform_task(shared_ptr<compaction_manager::task> task) {
+future<compaction_manager::task::validation_errors_stats> compaction_manager::perform_task(shared_ptr<compaction_manager::task> task) {
     _tasks.push_back(task);
     auto unregister_task = defer([this, task] {
         _tasks.remove(task);
     });
     cmlog.debug("{}: started", *task);
 
     try {
-        co_await task->run();
+        const auto &res = co_await task->run();
```

Comment:
nit: `auto res =` is simpler and more visible that you're storing the result in the automatic variable.
Then co_return res will move it (using copy elision)

`auto&& res =` is also good as a universal reference.  it might save the move here.

In any case, in our coding style, the `&` sticks to `auto`, so what you wrote should be `const auto& res =`